### PR TITLE
Python-version-specific install path for chpl-venv

### PR DESCRIPTION
**Pull request by awallace-cray**

### CODE REVIEW

**Comment by awallace-cray** [comment]
@ronawho, please have a look.  Tested manually on kay, jupiter, chap01, stpwchap

File: Makefile.devel [hunk 1]
```diff
@@ -15,6 +15,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+PYTHON_VERSION_DIR = $(shell python -c "from __future__ import print_function; \
```
Comment (ronawho) [inline Makefile.devel]: Should be able to use the `util/chplenv/chpl_python_version.py` script here too. Makefile.devel is only ever used by the top level makefile which sets CHPL_MAKE_HOME

**Comment by awallace-cray** [comment]
more testing tomorrow

File: third-party/chpl-venv/Makefile [hunk 2]
```diff
@@ -5,15 +5,13 @@ endif
 CHPL_MAKE_HOST_TARGET = --host
 include $(CHPL_MAKE_HOME)/make/Makefile.base
 
-PYTHON = $(shell which python)
+# do not include ./Makefile.include from Makefile.base
```
Comment (ronawho) [inline third-party/chpl-venv/Makefile]: This comment needs to be expanded to say why Makefile.include isn't included in Makefile.base

File: make/Makefile.base [hunk 3]
```diff
@@ -156,9 +156,9 @@ include $(THIRD_PARTY_DIR)/gmp/Makefile.include
 include $(THIRD_PARTY_DIR)/hwloc/Makefile.include
 include $(THIRD_PARTY_DIR)/re2/Makefile.include
 include $(THIRD_PARTY_DIR)/llvm/Makefile.include-$(CHPL_MAKE_LLVM)
-include $(THIRD_PARTY_DIR)/chpl-venv/Makefile.include
 include $(THIRD_PARTY_DIR)/fltk/Makefile.include
 include $(THIRD_PARTY_DIR)/libunwind/Makefile.include
+# do not include $(THIRD_PARTY_DIR)/chpl-venv/Makefile.include from Makefile.base
```
Comment (ronawho) [inline make/Makefile.base]: I don't think this comment should here

File: third-party/Makefile [hunk 4]
```diff
@@ -3,6 +3,9 @@ export CHPL_MAKE_HOME=$(shell pwd)/..
 endif
 include $(CHPL_MAKE_HOME)/make/Makefile.base
 
+# do not include chpl-venv/Makefile.include from Makefile.base
```
Comment (ronawho) [inline third-party/Makefile]: I'd remove this comment, and put the include directly under the Makefile.base include

File: third-party/chpl-venv/Makefile (hunk 2 above)
Comment (ronawho) [inline third-party/chpl-venv/Makefile]: Or I'd remove this comment, and instead stuff a comment in chpl-venv/Makefile.include about why it isn't/shouldn't be included in Makefile.base

File: third-party/Makefile [hunk 5]
```diff
@@ -3,6 +3,9 @@ export CHPL_MAKE_HOME=$(shell pwd)/..
...
+include chpl-venv/Makefile.include
```
Comment (ronawho) [inline third-party/Makefile]: This should be an absolute include like `include $(CHPL_HOME)/third-party/chpl-venv/Makefile.include`

**Comment by awallace-cray** [comment]
Rebased. Review comments addressed.

File: util/chplenv/__init__.py [hunk 6]
```diff
@@ -4,6 +4,7 @@
     'chpl_aux_filesys',
     'chpl_make',
     'chpl_platform',
+    'chpl_python_version',
```
Comment (ronawho) [inline util/chplenv/__init__.py]: This should be put down with the `General purpose helpers` instead of the core chplenv scripts

**Comment by ronawho** [comment]
Looks good to me,  I left one final comment about the order or chpl_python_version in __init__.py.

@benharsh mind giving this a second set of eyes?

**Comment by benharsh** [comment]
That addition in Makefile.devel will not add overhead to a program's compilation time, right?

If so, looks good to me! 👍

**Comment by awallace-cray** [comment]
Cancelled partially-finished Travis builds. Merging.

**Pull request merged**